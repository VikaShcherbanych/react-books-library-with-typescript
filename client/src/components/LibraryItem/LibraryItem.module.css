.bookCardWrap {
    padding-left: 2px;
    padding-top: 5px;
    padding-bottom: 6px;
    width: 100%;
    max-width: 280px;
    border-radius: 5px;
    background-color: #f4f2eb;
    transform: scale(1);
  }
  
  .bookCardLink {
    display: block;
    cursor: pointer;
  }
  
  .bookCardPoster {
    width: 260px;
    height: 340px;
    border-radius: 5px;
    transform: scale(1);
  }
  
  .bookCardWrap:hover {
    transform: scale(1.05);
  }
  
  .bookCardTitle {
    display: block;
    padding: 5px 10px;
    margin: 0;
    text-transform: uppercase;
    font-size: 15px;
    font-weight: 400;
    line-height: 1.333;
    color: #000;
  }
  
  .bookCardAuthor {
    display: block;
    font-size: 14px;
    font-weight: 400;
    line-height: 1.333;
    color: #000;
  }
  
  .bookCardCategory {
    height: 40px;
    align-items: center;
    padding: 7px;
    font-size: 12px;
    color: #000;
  }
  
  @media screen and (max-width: 767px) {
    .bookCardWrap:not(:last-child) {
      margin-bottom: 20px;
    }
  }
  
  @media screen and (min-width: 768px) {
    .bookCardPoster {
      width: 276px;
    }
  }
  
  @media screen and (min-width: 1024px) {
    .bookCardPoster {
      width: 266px;
    }
  }