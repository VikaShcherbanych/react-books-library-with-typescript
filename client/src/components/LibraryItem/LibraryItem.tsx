import React from 'react';
import { toast } from 'react-toastify';

import Button from '../../components/Button/Button';
import { IBookFromLibrary } from '../../models/IBookFromLibrary';
import { deleteBook } from '../../redux/reducers/library/BookActionCreators';
import { useAppSelector, useAppDispatch } from '../../hooks/redux';
import s from './LibraryItem.module.css';

interface IPropBook {
  book: IBookFromLibrary;
}

const LibraryItem: React.FC<IPropBook> = ({ book }: IPropBook) => {
  const dispatch = useAppDispatch();
  const store = useAppSelector(state => state);
  const handleDeleteBookFromLibrary = (id: string) => {
    dispatch(deleteBook(id));
    if (!store.library.error) {
      toast.success('Book deleted!');
    }
  };
  return (
    <li className={s.bookCardWrap} key={book.id}>
      <a href={book.previewLink} className={s.bookCardLink}>
        <img
          src={book.imageLink}
          alt={book.title}
          className={s.bookCardPoster}
          width="260"
          height="340"
        />
      </a>
      <h2 className={s.bookCardTitle}>
        {book.title.length > 27 ? `${book.title.slice(0, 25)}...` : book.title}
      </h2>
      <p className={s.bookCardAuthor}>
        {book.authors.length > 0
          ? book.authors[0].slice(0, 25)
          : 'The author is unknown'}
      </p>
      <p className={s.bookCardCategory}>
        Category:{' '}
        {book.categories.length > 0 ? book.categories[0] : 'no category'}
      </p>
      <Button
        onClick={() => handleDeleteBookFromLibrary(book.id)}
        text="Delete from library"
      />
    </li>
  );
};
export default LibraryItem;
