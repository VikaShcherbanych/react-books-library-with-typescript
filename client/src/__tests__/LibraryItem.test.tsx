import React from "react";
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import { toast } from "react-toastify";
import  LibraryItem  from "../components/LibraryItem/LibraryItem";

configure({adapter: new Adapter()});

jest.mock("../images/not-found-img.jpeg", () => ({}));
jest.mock("../components/LibraryItem/LibraryItem.module.css", () => ({}));
jest.mock('../components/Button/Button.tsx', () => ({}));
toast.success = jest.fn();

jest.mock('../hooks/redux', () => ({
  useAppDispatch: () => {},
  useAppSelector: () => ({
    library: {}
  }),
}));

const expectedBook = {
      id: 'gwCrDwAAQBAJ',
      title: "A Cat's Tale",
      authors: ['Dr. Paul Koudounaris', 'Baba the Cat'],
      categories: ['Fantasy', 'Cat'],
      imageLink: 'http://books.google.com/books/content?id=gwCrDwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api',
      previewLink: 'http://books.google.com/books?id=gwCrDwAAQBAJ&printsec=frontcover&dq=cats&hl=&cd=1&source=gbs_api',
};

describe("BookItem", () => {
    test("should show BookItem", () => {
        const sut = shallow(<LibraryItem book={expectedBook}/>);
        expect(sut).toMatchSnapshot();
    });
});