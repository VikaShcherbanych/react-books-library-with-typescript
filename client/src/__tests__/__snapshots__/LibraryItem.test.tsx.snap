// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookItem should show BookItem 1`] = `ShallowWrapper {}`;
